api/watch/funcs.go

```diff
@@ -204,11 +211,41 @@ func checksWatch(params map[string]interface{}) (WatcherFunc, error) {
 		if err != nil {
 			return nil, nil, err
 		}
+		if len(tags) > 0 {
+			checks = filterChecksByTags(checks, tags)
+		}
+
```

Comment:
The `/v1/agent/checks` API supports bexpr filtering and I'm assuming this `health.Checks()` method uses that underlying API. If that's the case, I think it would be more efficient to filter the response server-side using a filter like `tags in ServiceTags`.

Have you explored this approach and is it possible to implement?

Reply:
This sounds a better solution. Let me dig into that. Thanks, @blake

Reply:
@blake , done. Move the filter to server side by passing a `bexpr` filter.